corral/cmd/cmd_update.pony

```diff
@@ -61,25 +61,18 @@ actor _Updater
     try
       (let deptag, let dep) = updates.remove(dep')?
 
+      // TODO: consider parsing version much earlier, maybe in Bundle.
+      // TODO: also consider allowing literal tags and not just constraints
+      // expressions.
+      let constraints = Constraints.parse(dep.data.version)
+
       ctx.log.fine(
         "tags for [" + updates.size().string() + "] " + dep.locator.string()
           + ": " + tags.size().string())
-      let source: ss.InMemArtifactSource = source.create()
 
-      for tg in tags.values() do
-        ctx.log.fine("  tag:" + tg)
-        let artifact = ss.Artifact("A", sv.ParseVersion(tg))
-        source.add(artifact)
-      end
-      ctx.log.fine("")
+      let result = Constraints.solve(constraints, tags, ctx.log)
 
-      // TODO: consider parsing version much earlier, before we even get here.
-      // TODO: also consider allowing literal tags and not just constraints
-      // expressions.
-      let constraints = parse_constraints(dep.data.version)
-      let result = ss.Solver(source).solve(constraints.values())
-
-      ctx.log.fine("result err: " + result.err)
+      // TODO: probably OK to have multiple solutions: choose 'best' with a strategy like 'latest'.
```

Comment:
I don't understand this TODO or why it's here.

I'm going to stop commenting on the TODO's but they all feel like they should be issues. I wouldn't be able to pick up and do work based on any of these TODO's.

Reply:
This one is related to the solver returning more than one solution to a version constraint. Many dependency tools just pick the latest solution, but we could have other options.

I'll create an issue. Would you like me to add links to the TODO issues in the TODO comments?

Reply:
Re the question. 

Yes! In fact, while you were typing that, I was saying exactly that in a different place.

Reply:
I've updated a number of TODOs with issue links. I think I caught the important ones.